src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -476,6 +476,11 @@ async Task<NSUrlRequest> CreateRequest (HttpRequestMessage request)
 			};
 
 			if (stream != Stream.Null) {
+				// Rewind the stream to the beginning in case the HttpContent implementation
+				// will be accessed again (e.g. for retry/redirect) and it keeps its stream open behind the scenes.
+				if (stream.CanSeek)
+					stream.Seek(0, SeekOrigin.Begin);
```

Comment:
Why are you assuming that the thing I want to send is from the beginning of the stream? If I do a partial upload of a file (ie resumed upload), I'd have an offset in my filestream.

Reply:
I'm not sure that described behavior has been supported before. Do you have an example?

As I can see, the current implementation of `NSUrlSessionHandler` uses simple `NSURLSessionDataTask` under the hood. If you need specific upload a file you should be sure that the native layer supports this behavior.

There is [NSURLSessionUploadTask](https://developer.apple.com/documentation/foundation/nsurlsessionuploadtask?language=objc) in iOS for upload but we can't simply use it without reworking the native handler.

Reply:
> If you need specific upload a file you should be sure that the native layer

I don't see why I need to do that. The normal http handlers does this just fine. Regardless, you're posting some stream,. and it's wrong to assume it's from the beginning of that stream, and that was all I was pointing out.

Reply:
If that is true then the behaviour in the android equivalent of this class is wrong.  See: [AndroidMessageHandler.cs](https://github.com/xamarin/xamarin-android/blob/17c74c5cfcc383464931e6a2f828a689ba998965/src/Mono.Android/Xamarin.Android.Net/AndroidMessageHandler.cs#LL564C5-L565C40)

```
if (stream.CanSeek)
    stream.Seek (0, SeekOrigin.Begin);
```

Reply:
Yup it would seem so

Reply:
I think doing the same thing Android does is good enough for now, it's still better than the previous behavior.